codex-rs/tui/src/bottom_pane/mod.rs

```diff
@@ -148,19 +182,38 @@ impl BottomPane<'_> {
         }
     }
 
-    /// Update the status indicator text (only when the `StatusIndicatorView` is
-    /// active).
+    /// Update the status indicator text. Prefer replacing the composer with
+    /// the StatusIndicatorView so the input pane shows a single-line status
+    /// like: `▌ Working [·] waiting for model`.
     pub(crate) fn update_status_text(&mut self, text: String) {
-        if let Some(view) = &mut self.active_view {
-            match view.update_status_text(text) {
-                ConditionalUpdate::NeedsRedraw => {
+        if let Some(view) = self.active_view.as_mut() {
+            match view.update_status_text(text.clone()) {
+                bottom_pane_view::ConditionalUpdate::NeedsRedraw => {
                     self.request_redraw();
+                    return;
                 }
-                ConditionalUpdate::NoRedraw => {
-                    // No redraw needed.
-                }
+                bottom_pane_view::ConditionalUpdate::NoRedraw => {}
```

Comment:
This is the only case in which we don't `return;`, so maybe move the logic from "Fallback" into here and then eliminate the `return;` statements?